homework.js

```diff
@@ -0,0 +1,207 @@
+//Q1
+
+const  summation = function (n) {
+  let output = 0;
+  for( i = 1;  i <= n; i++) {
+    output = output  + i;
+  }
+  return output ;
+};
+
+console.log('Sum of numebers from 1 to '+ 5 +' is '+ summation(5));
+
+
+//Q2
+
+
+const largestEven = function (arr) {
+  const evens = [];
+
+  for (i =1; i < arr; i++) {
+    if(i % 2 === 0) {
+      evens.push(i);
+    }
+
+  }
+ 
+ function myFunc(total, num) {
+  return total + num;
+}
+  console.log(evens.reduce(myFunc))
+ 
+  }    
+  console.log(largestEven(5));
+
+
+ //Q3
+
+ 
+const getAvg = function (array) {
+let count =0;
+
+for (i = 0 ; i < array.length ; i++) 
+{
+
+count  += array[i] 
+ }
+return  Math.floor(count / array.length)
+}
+
+console.log(getAvg ([8, 4, 2, 4]))
+
+//Q4 
+
+function reverse (str) {
+
+const splitString = str.split("")
+
+
+const reverse = splitString.reverse(); 
+
+
+const joinArray = reverse.join("");
+
+return joinArray;
+}
+
+console.log(reverse('caterpillar'))
+
+//Q5
+
+const addDashes =['test1', 'test2', 'test3'] ;
+function combinesThemDash () { 
+  let dash ;
+for (i=0 ; i <= addDashes.length ; i++){
+if (i === addDashes.length)
+dash = addDashes.join("-")
+}
+return dash  ;
+};
+console.log(combinesThemDash());
+
+or 
+
+const addDashes= function(dash){
+return dash.join("-");
+
+}
+
+console.log(addDashes(['test1', 'test2', 'test3']));
+
+
+
+//Q6
+
+const printNumbers = function (num) {
+  const count = []
+  for (i=1 ; i<=num ; i++){
+   count.push(i)
+  }
+  for (j=num-1 ; j > 0 ; j--){
+   count.push(j)
+  
+  }
+return count.join(' ')
+}
+
+printNumbers(3)
+
+
+
+//Q7
+
+function find(LetterA){
+  let FindA = 'a';
+  FindA = FindA.split('');
+return LetterA.filter(function(LetterA){
+return FindA.every(function(char){
+return LetterA.includes(char);
+});
+});
+}
+
+console.log(find(['cat', 'rabbit', 'dog', 'frog']));
+
+
+//Q8
+
+function find(wordsWithLetter){
```

Comment:
functionName (firstArgument, secondArgument)
This function should have 2 arguments the first is a letter, the second is an array of strings.